modules/platforms/dotnet/Apache.Ignite/Internal/Table/RecordBinaryView.cs

```diff
@@ -0,0 +1,688 @@
+
+            lock (_latestSchemaLock)
+            {
+                if (schemaVersion > _latestSchemaVersion)
+                {
+                    _latestSchemaVersion = schemaVersion;
+                }
+            }
+
+            return schema;
+        }
+
+        private void WriteTuple(
+            PooledArrayBufferWriter buf,
+            Transactions.Transaction? tx,
+            Schema schema,
+            IIgniteTuple tuple,
+            bool keyOnly = false)
+        {
+            var w = buf.GetMessageWriter();
+
+            WriteTupleWithHeader(ref w, tx, schema, tuple, keyOnly);
+
+            w.Flush();
+        }
+
+        private void WriteTuples(
+            PooledArrayBufferWriter buf,
+            Transactions.Transaction? tx,
+            Schema schema,
+            IIgniteTuple t,
+            IIgniteTuple t2,
+            bool keyOnly = false)
+        {
+            var w = buf.GetMessageWriter();
+
+            WriteTupleWithHeader(ref w, tx, schema, t, keyOnly);
+            WriteTuple(ref w, schema, t2, keyOnly);
+
+            w.Flush();
+        }
+
+        private void WriteTuples(
+            PooledArrayBufferWriter buf,
+            Transactions.Transaction? tx,
+            Schema schema,
+            IEnumerator<IIgniteTuple> tuples,
+            bool keyOnly = false)
+        {
+            var w = buf.GetMessageWriter();
+
+            w.Write(_table.Id);
+            WriteTx(ref w, tx);
+            w.Write(schema.Version);
+            w.Flush();
+
+            var count = 0;
+            var countPos = buf.ReserveInt32();
+
+            do
+            {
+                var tuple = tuples.Current;
+
+                if (tuple == null)
+                {
+                    throw new ArgumentException("Tuple collection can't contain null elements.");
+                }
+
+                WriteTuple(ref w, schema, tuple, keyOnly);
+                count++;
+            }
+            while (tuples.MoveNext()); // First MoveNext is called outside to check for empty IEnumerable.
+
+            buf.WriteInt32(countPos, count);
+
+            w.Flush();
+        }
+
+        private void WriteTupleWithHeader(
+            ref MessagePackWriter w,
+            Transactions.Transaction? tx,
+            Schema schema,
+            IIgniteTuple tuple,
+            bool keyOnly = false)
+        {
+            w.Write(_table.Id);
+            WriteTx(ref w, tx);
+            w.Write(schema.Version);
+
+            WriteTuple(ref w, schema, tuple, keyOnly);
+        }
+
+        private ValueTask<ClientSocket> GetSocket(Transactions.Transaction? tx)
+        {
+            if (tx == null)
+            {
+                return _table.Socket.GetSocketAsync();
+            }
+
+            if (tx.FailoverSocket != _table.Socket)
+            {
+                throw new IgniteClientException("Specified transaction belongs to a different IgniteClient instance.");
+            }
```

Comment:
So we only use `FailoverSocket` property to identify transaction's belonging to the client?

Reply:
Yes. Maybe not perfect, but I don't have better ideas.

In fact, we can remove this check and the code will still work, because transaction has it's own socket. But I don't think we should allow this scenario.

Reply:
No, I think this solutions is OK, I just wanted to make sure.